### DISCUSSION TITLE
Auto Hide Properties

### DISCUSSION THREAD

**Discussion by Moyf** [chrisgurney/obsidian-note-toolbar#154] [Ideas]
This issue may not closely related to Toolbar, but since NTB itself has those commands - I've decided to bring it up 😄
![image](https://github.com/user-attachments/assets/194c5f78-bdb4-4f92-8800-3459c2cc88a0)

So, is it possible to automatically fold the Property when switching tabs/opening tabs?
Essentially making "Folded Properties" the default behavior!

---

In Obsidian's settings, you can set the default to show/hide Property, but there's no way to set it to collapse by default. 
![image](https://github.com/user-attachments/assets/2732ab60-5867-4665-a778-9c401ae20275)

If there are other ways to achieve this without embedding NTB, I would greatly appreciate any guidance!

**Comment by chrisgurney** [comment]
Thanks @Moyf!

I had looked into the possibility of running commands on open a while back and [hit a dead end](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/23#discussioncomment-9060546).

It might be worth another shot, however.

I also wanted to improve the Properties commands by actually updating the local storage state (where fold status is tracked), but IIRC I think that will only help on a note-by-note basis. 

I'll start there and then see if anything else can be done.

**Reply by Moyf** [reply]
> Thanks @Moyf!
> 
> I had looked into the possibility of running commands on open a while back and [hit a dead end](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/23#discussioncomment-9060546).
> 
> It might be worth another shot, however.
> 
> I also wanted to improve the Properties commands by actually updating the local storage state (where fold status is tracked), but IIRC I think that will only help on a note-by-note basis.
> 
> I'll start there and then see if anything else can be done.

I seem to have seen it on some plugin before, which binds a listener event to the default **tab switching** behavior of Obsidian, so that it can do something when the focus of the tab changes...  
Perhaps a similar implementation can be referenced?  

Just found some possible references:  
https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Clean+up+resources+when+plugin+unloads  
https://docs.obsidian.md/Reference/TypeScript+API/EventRef  

My guess:  
Perhaps the event registration could be on [on active leaf change](https://docs.obsidian.md/Reference/TypeScript+API/Workspace/on('active-leaf-change'))?

**Reply by chrisgurney** [reply]
@Moyf thanks! I'll take a look when I get a chance.

**Comment by Moyf** [comment]
https://github.com/Benature/obsidian-metadata-hider
Just for your information, this plugin could achive what I want (auto fold properties)! Maybe there could be some clues 😄

**Reply by Moyf** [reply]
It seems that it's binding event on "file-open":
```js
this.registerEvent(this.app.workspace.on('file-open', (file) => {
	if (!this.settings.autoFold) { return; }
	const metadataElement = document.querySelector('.workspace-leaf.mod-active .metadata-container');
	if (!metadataElement?.classList.contains('is-collapsed')) {
		// @ts-ignore
		this.app.commands.executeCommandById(`editor:toggle-fold-properties`);
	}
}));
```

**Comment by chrisgurney** [comment]
@Moyf Properties should now stay hidden across tabs, as of [`v1.23-beta-03`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.23-beta-03). This has nothing to do with Obsidian's setting; you'll have to run Note Toolbar's _Toggle Properties_ or _Hide Properties_ command first, and then it should be remembered on that device.

Give it a try and let me know how it works for you!

**Reply by Moyf** [reply]
I've tried it, it works!
However, could the `Fold Properties` also supports this feature? The difference is that, for `folded properties`, you can use default hotkeys (`Adding property`, `Fold/unfold properties` commands ) to make it show/expand, but for `hidden properties` it seems that only `show properties` command can make it appear again.

**Reply by chrisgurney** [reply]
@Moyf I'm not fully sure what you mean. And do you mean the `Fold properties` command provided by Note Toolbar?

**Reply by Moyf** [reply]
> @Moyf I'm not fully sure what you mean. And do you mean the `Fold properties` command provided by Note Toolbar?

Yes!

**Reply by Moyf** [reply]
> @Moyf I'm not fully sure what you mean.

🤔 Well, I mean now each page can auto hide the property, then could each page also "Auto Fold Property"?
Like when you switch or open a new tab, auto fold the property of the page.

**Reply by chrisgurney** [reply]
@Moyf Properties should now stay folded across tabs, as of [`1.23-beta-08`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.23-beta-08). As before, this has nothing to do with Obsidian's setting; you'll have to run Note Toolbar's _Fold Properties_ command first, and then it should be remembered on that device.

**Reply by Moyf** [reply]
> @Moyf Properties should now stay folded across tabs, as of [`1.23-beta-08`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.23-beta-08). As before, this has nothing to do with Obsidian's setting; you'll have to run Note Toolbar's _Fold Properties_ command first, and then it should be remembered on that device.

```ignore
Perhaps I’ve realized what I didn’t express clearly—  
What I actually want is for notes to automatically collapse on every page, rather than only remembering the ones I manually collapsed.  
For example, if I currently have 5 tabs, I’d like all 5 tabs to be in a folded state after using the "fold properties" command — until I explicitly expand them with a command.  

I’m not sure if this is the intended design, but currently, after I fold one note property, the other 4 note properties remain unfolded.  

(Though I should mention, there are existing plugins—as far as I know, two of them, and one is already available in community plugins : https://github.com/tbergeron/obsidian-fold-properties-by-default.)  
So here, I’m just trying to clarify my idea, but I don’t think this development request is of high priority.  

If you feel the current behavior is sufficient and aligns with the expected design, then you can close this Issue… Oh wait, this is the discussion —so you can just ignore this suggestion!  😛
```
~~IGNORE ABOVE~~

Oh wait, you mean `beta-08`!
I'm still in `beta-07` 😂
I'll try 08 then! Thank you!

**Reply by Moyf** [reply]
In version 08, it works, although there is a noticeable delay when switching pages. 🤔

**Reply by chrisgurney** [reply]
> although there is a noticeable delay when switching pages

@Moyf Do you mean when clicking into tabs/views? Do you have multiple views open?

**Reply by Moyf** [reply]
> > although there is a noticeable delay when switching pages
> 
> @Moyf Do you mean when clicking into tabs/views? Do you have multiple views open?

I thought I replied to this before 😲

I encounter this issue in two scenarios:  
1. When creating a new note  
2. When switching to a different tab for the first time  

That’s how it happens:
https://github.com/user-attachments/assets/48d177a5-f3a3-41d7-a2bf-c2dd6cda0146

![image](https://github.com/user-attachments/assets/f4d6a9ae-80fa-4241-9a7b-e3ef336ed72e)

> If a more complex Theme is used, this delay will become even more pronounced.

**Comment by chrisgurney** [comment]
Marking this as completed. Can address issues as they come up.

**Comment by chrisgurney** [comment]
@Moyf FYI I've now made this _auto-hide_ behavior a setting as of [`1.23.09`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.23.09), as it wasn't clear to at least one user that's what was happening. It's disabled by default, so you'll need to enable it.

**Reply by Moyf** [reply]
Roger that, thank you!
